### PR TITLE
[SPARK-122] Adjust Galaxy's search parameters to use API indexes

### PR DESCRIPTION
**Pull request by Eearslya**
Galaxy was performing searches for system main stars by using the `systemId64` and `isMainStar` properties, which are not indexed. This caused slow and inefficient searches for the API.

This changes the search to use the `systemId` property, which is properly indexed. It also adjusts the code to find the Main star by itself, which should be relatively lightweight.

### CODE REVIEW

**Comment by Holladiewal** [comment]
I think the BugFix label is more appropriate here, as refactoring is changing the structure of the code without changing it's (external) behaviour. Since this changes the requests that are send to the API, it is explicitly changing behaviour which is out of scope for a refactor.

**Comment by shatteredbeam** [comment]
I'm Okay with refactor.  This is more of an optimization pass than a direct bugfix.

File: src/packages/galaxy/galaxy.py [hunk 1]
```diff
@@ -80,15 +80,34 @@ def __init__(self, url: str = None):
...
-            if main_star['meta']['results']['available'] > 0:
-                sys['spectral_class'] = main_star['data'][0]['attributes']['subType'][0]
+            main_star = await self._find_main_star(system_id)
+            sys['spectral_class'] = main_star['spectral_class'] if main_star is not None else None
             return StarSystem(position=Vector(**sys['coords']),
                               name=sys['name'],
-                              spectral_class=sys.get('spectral_class'))
+                              spectral_class=sys['spectral_class'])
+
+    async def _find_main_star(self, system_id: int) -> Optional[Dict]:
```
Comment (shatteredbeam) [inline src/packages/galaxy/galaxy.py:91]: [Nitpick] Does this *actually* return None if the system isn't found?  Where does that happen?

**Review by shatteredbeam — CHANGES_REQUESTED** [review]
I think you may want to make sure that your method explicitly returns None to prevent an error case, but other than that looks good.  I'm okay with this not being explicitly set *if* you can demonstrate that the method can ONLY return the intended type or None.

File: src/packages/galaxy/galaxy.py (hunk 1 above)
Comment (Eearslya) [reply src/packages/galaxy/galaxy.py:91]: The `return None` happens implicitly with Python if the function reaches the end without an explicit `return`.

This case is also tested with `test_find_system_by_invalid_name`, asserting the result is `None`.

Comment (shatteredbeam) [reply src/packages/galaxy/galaxy.py:91]: Good enough for me.  Thanks.

**Review by shatteredbeam — DISMISSED** [review]

**Review by theunkn0wn1 — APPROVED** [review]
LGTM
